app/models/building.rb

```diff
@@ -0,0 +1,8 @@
+class Building < ActiveRecord::Base
+  has_many :tenants, class_name: User.name
```

Comment:
Good question, and the unknowns are a big reason for why I decided to hold off on implementing the Apartment relationship.

I think one way to represent could be something like:

```ruby
class Building
  has_many :tenants, through: :apartments

class Apartment
  belongs_to :building
  has_many :tenants

class Tenant
  belongs_to :apartment
```

Then some questions arise such as:
- Should there to be more than one Tenant per Apartment allowed at a time (like each member of a household or roommates)?
- If there are multiple Tenants in an Apartment, how should we log that as the Tenants in the Reading? Do we need to store an array of tenant_ids there or will just the apartment_id suffice? Maybe a lookup can be done by a record that's like: tenant_id, apartment_id, residency_start, residency_end to associate with a reading. Figuring out how we need to present the information will help with figuring out how to store the data optimally.
- Should a Tenant essentially be a child of the User class or should it represent something in itself?  Does every Tenant need to be a User account or are User accounts only for people who need to log into the app?  For example, you could create a Tenant object after installing a sensor, but if the person never logs into the app, they never have a User account.
- If we want Tenants to be a type of User account, maybe Single Table Inheritance could be used here... I'm not a big fan of Rails STI though because of various issues I've run into while using it.